inc/Widget/SmailyWidget.php

```diff
@@ -210,10 +460,39 @@ class="widefat"
+		$instance['form_layout']                       = isset( $new_instance['form_layout'] ) ? sanitize_text_field( $new_instance['form_layout'] ) : '';
+		$instance['autoresponder']                     = isset( $new_instance['autoresponder'] ) ? sanitize_text_field( $new_instance['autoresponder'] ) : '';
+		$instance['email_field_placeholder']           = isset( $new_instance['email_field_placeholder'] ) ? sanitize_text_field( $new_instance['email_field_placeholder'] ) : '';
+		$instance['name_field_placeholder']            = isset( $new_instance['name_field_placeholder'] ) ? sanitize_text_field( $new_instance['name_field_placeholder'] ) : '';
+		$instance['submit_button_text']                = isset( $new_instance['submit_button_text'] ) ? sanitize_text_field( $new_instance['submit_button_text'] ) : '';
+		$instance['submit_button_color']               = isset( $new_instance['submit_button_color'] ) ? sanitize_text_field( $new_instance['submit_button_color'] ) : '';
+		$instance['submit_button_text_color']          = isset( $new_instance['submit_button_text_color'] ) ? sanitize_text_field( $new_instance['submit_button_text_color'] ) : '';
+		$instance['use_site_submit_button_color']      = isset( $new_instance['use_site_submit_button_color'] ) ? (bool) (int) sanitize_text_field( $new_instance['use_site_submit_button_color'] ) : false;
+		$instance['use_site_submit_button_text_color'] = isset( $new_instance['use_site_submit_button_text_color'] ) ? (bool) (int) sanitize_text_field( $new_instance['use_site_submit_button_text_color'] ) : false;
+
+		// Validate that input isn't empty, set default value, if it is.
+		$instance['title']                   = ( empty( $instance['title'] ) ) ? '' : $instance['title'];
+		$instance['form_layout']             = ( empty( $instance['form_layout'] ) ) ? 'layout-1' : $instance['form_layout'];
+		$instance['email_field_placeholder'] = ( empty( $instance['email_field_placeholder'] ) ) ? '' : $instance['email_field_placeholder'];
+		$instance['name_field_placeholder']  = ( empty( $instance['name_field_placeholder'] ) ) ? '' : $instance['name_field_placeholder'];
+		$instance['submit_button_text']      = ( empty( $instance['submit_button_text'] ) ) ? __( 'Send', 'smaily' ) : $instance['submit_button_text'];
+
+		// If button color isn't set, check default checkbox.
+		if ( $instance['use_site_submit_button_color'] === true ) {
+			$instance['use_site_submit_button_color'] = true;
+		} elseif ( empty( $instance['submit_button_color'] ) ) {
+			$instance['use_site_submit_button_color'] = true;
+		}
+
+		// If text color isn't set, check default checkbox.
+		if ( $instance['use_site_submit_button_text_color'] === true ) {
+			$instance['use_site_submit_button_text_color'] = true;
+		} elseif ( empty( $instance['submit_button_text_color'] ) ) {
+			$instance['use_site_submit_button_text_color'] = true;
+		}
```

Comment:
Since you are using boolean value now, this part can be simplified to:

```suggestion
		// If button color isn't set, check default checkbox.
		if ( empty( $instance['submit_button_color'] ) ) {
			$instance['use_site_submit_button_color'] = true;
		}

		// If text color isn't set, check default checkbox.
		if ( empty( $instance['submit_button_text_color'] ) ) {
			$instance['use_site_submit_button_text_color'] = true;
		}
```

because when casting input value through `(bool) (int)` your only possible options will always be either `true` or `false`.